04_numeronym/patrickmarabeas/main.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// Convert string to a numeronym
+func generateNumeronym(s string) string {
+	length := len(s)
+	// Early bailout if string is less than 4 characters
+	if length < 4 {
+		return s
+	}
+	// Format as "{unicode character}{decimal}{unicode character}"
+	return fmt.Sprintf("%c%d%c", s[0], length-2, s[length-1])
+}
+
+// Convert n strings into a slice of numeronyms
+func numeronyms(vals ...string) []string {
+	result := make([]string, len(vals))
+	i := 0
+	for _, numeronym := range vals {
```

Comment:
Wouldn't need to declare i separately this way, `for i, numeronym := range vals {`

Reply:
Good point! Was in the mindset of looping maps : )